torch/quantization/_equalize.py

```diff
@@ -0,0 +1,130 @@
+import torch
+import copy
+
+def max_over_ndim(input, axis_list, keepdim=False):
+    ''' Applies 'torch.max' over the given axises
+    '''
+    axis_list.sort(reverse=True)
+    reduced_input = input
+    for axis in axis_list:
+        reduced_input, _ = reduced_input.max(axis, keepdim)
+    return reduced_input
+
+def min_over_ndim(input, axis_list, keepdim=False):
+    ''' Applies 'torch.min' over the given axises
+    '''
+    axis_list.sort(reverse=True)
+    reduced_input = input
+    for axis in axis_list:
+        reduced_input, _ = reduced_input.min(axis, keepdim)
+    return reduced_input
+
+def channel_range(input, axis=0):
+    ''' finds the range of weights associated with a specific channel
+    '''
+    size_of_tensor_dim = len(input.size())
+    mins = input
+    maxs = input
+
+    axis_list = list(range(size_of_tensor_dim))
+    axis_list.remove(axis)
+
+    mins = min_over_ndim(mins, axis_list)
+    maxs = max_over_ndim(maxs, axis_list)
+
+    assert(mins.size()[0] == input.size()[axis])
+    return maxs - mins
+
+def cross_layer_equalization(module1, module2, output_axis=0, input_axis=1):
+    ''' Given two adjacent tensors', the weights are scaled such that
+    the ranges of the first tensors' output channel are equal to the
+    ranges of the second tensors' input channel
+
+    '''
+    _supported_types = [torch.nn.Conv2d, torch.nn.Linear]
```

Comment:
We should probably move this declaration to the top of the file. That way the developers will know exactly what is supported without a need to search for it.